sarracenia/moth/amqp.py

```diff
@@ -132,10 +135,20 @@ def _msgRawToDict(self, raw_msg) -> sarracenia.Message:
                 '%23', '#').replace('%22', '*')
             msg['exchange'] = raw_msg.delivery_info['exchange']
             if self.o['sourceFromExchange']:
-                source = self.o.get_source_from_exchange(msg['exchange'])
-                if source:
-                    msg['source'] = source
-                    msg['_deleteOnPost'] |= set(['source'])
+                itisthere = re.match( "xs_([^_]+)_.*", msg['exchange'] )
+                if itisthere:
+                    source = itisthere[1]
+                else:
+                    itisthere = re.match( "xs_([^_]+)", msg['exchange'] )
+                    if itisthere:
+                        source = itisthere[1]
+            else:
+                if 'source' in self.o:
+                    source = self.o['source']
+
+            if source:
```

Comment:
I think this could fail with `NameError: name 'source' is not defined` when

1. `source` is not in `self.o`
2. and `sourceFromExchange` is `False`

or 

1. `sourceFromExchange` is `True`, but both regex matches fail to determine the source, which would happen when the exchange does not start with `xs_` (e.g. xpublic).

Reply:
resolved with new commit...